jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -712,12 +712,12 @@ public void interrupt() {
 		sun.nio.ch.Interruptible localBlockOn = blockOn;
 		if (localBlockOn != null) {
 			localBlockOn.interrupt(this);
-        }
-        /*[IF Java14]*/
-        if (!isAlive()) {
+		}
+		/*[IF Java14]*/
+		if (threadRef == NO_REF) {
 		deadInterrupt = true;
```

Comment:
Please indent this line properly with this change.

Reply:
I'm also still not convinced this is correct on 32-bit platforms - if there's no lock or volatile on the `threadRef` access, there is the possibility of tearing (may not actually be possible, depending how the macros work).

Reply:
@pshipton,   do we still need to add `volatile` to threadRef to ensure the safety when accessing `threadRef`?

Reply:
The code is inside synchronized(lock) which protects threadRef, this change doesn't require adding volatile.

Reply:
The original incorrect indentation made it look like the code wasn't inside the synchronized(lock), but it is.

Reply:
Similarly to the other change to check isAlive, there should be a comment that a method call is being avoided.